### PR TITLE
syncer: fix skip query

### PR DESCRIPTION
**Pull request by nolouch**
@GregoryIan @andelf PTAL

### CODE REVIEW

File: syncer/filter.go [hunk 1]
```diff
@@ -61,6 +62,22 @@ func (s *Syncer) skipQueryEvent(sql string, schema string) bool {
...
+
+	if strings.HasPrefix(sql, "OPTIMIZE TABLE") {
+		return true
+	}
+
+	if strings.HasPrefix(sql, "DROP TRIGGER") {
+		return true
+	}
+
+	r := regexp.MustCompile(`(^CREATE (DEFINER=\S+ )*TRIGGER)`)
```
Comment (GregoryIan) [inline syncer/filter.go]: we can make it a global variable

File: syncer/filter.go [hunk 2]
```diff
@@ -61,6 +62,22 @@ func (s *Syncer) skipQueryEvent(sql string, schema string) bool {
...
+	if r.FindStringIndex(sql) != nil {
```
Comment (GregoryIan) [inline syncer/filter.go]:
```
indics := r.FindStringIndex(sql)
if len(indics) >0 && indics [0] == 0 {
    return true
}
```

**Comment by andelf** [comment]
LGTM

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**